cron/workspacesTransition.js

```diff
@@ -0,0 +1,111 @@
+const AWS = require('aws-sdk');
+const { callbackify } = require('util');
+
+const logger = require('../lib/logger');
+const config = require('../lib/config');
+const request = require('request-promise-native');
+const async = require('async');
+
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports = {};
+
+module.exports.run = callbackify(async() => {
+    if (!config.runningInEc2) return;
+
+    await checkDBConsistency();
+    await terminateHosts();
+    await checkHealth();
+});
+
+async function checkDBConsistency() {
+    /* Attempt to make the list of hosts in EC2 consistent with what
+       we see in the database. */
+
+    const ec2 = new AWS.EC2();
+    const running_host_set = new Set();
+    const reservations = (await ec2.describeInstances({
+        Filters: [
+            {
+                Name: 'tag-key',
+                Values: [config.workspaceLoadLaunchTag],
+            },
+            {
+                Name: 'instance-state-name',
+                Values: [ 'pending', 'running' ],
+            },
+        ],
+        MaxResults: 500,
+    }).promise()).Reservations;
+    for (const reservation of reservations) {
+        for (const instance of Object.values(reservation.Instances)) {
+            running_host_set.add(instance.InstanceId);
+        }
+    }
+
+    const db_hosts = new Set((await sqldb.queryAsync(sql.select_nonterminated_or_nonterminating_workspace_hosts, [])).rows.map(instance => instance.instance_id));
+    const db_hosts_terminating = new Set((await sqldb.queryAsync(sql.select_running_or_terminating_workspace_hosts, [])).rows.map(instance => instance.instance_id));
```

Comment:
For consistency with the query name:

```suggestion
    const db_hosts_running_or_terminating = new Set((await sqldb.queryAsync(sql.select_running_or_terminating_workspace_hosts, [])).rows.map(instance => instance.instance_id));
```